data/fetcher/fetcher.go

```
@@ -529,6 +529,18 @@ func (self *Fetcher) PersistSnapshot(
 	}
 	// note: only update status when it's pending status
 	snapshot.ExchangeBalances = allEBalances
+
+	// persist blockchain balance
+	// if blockchain balance is not valid then auth snapshot will also not valid
+	for _, balance := range bbalances {
+		if !balance.Valid {
+			snapshot.Valid = false
+			if balance.Error != "" {
```

so, what happens if balance.Error is not empty? the error is ignored?

---

I thought it would be non-sense to add an empty string. I added the error message.